### ISSUE TITLE
Implement new tool `osrm-customize`

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#3350]
OSRM is going to grow a new speedup approach - MLD.  https://github.com/Project-OSRM/osrm-backend/projects/3

We will need a new pre-processing tool - `osrm-customize`.  This tool will:

  - take the edge-based-graph
  - take the partition data generated by https://github.com/Project-OSRM/osrm-backend/issues/3205
  - import traffic data into the edge-based-graph
  - calculate an MLD overlay
  - serialize it to disk

Basically, this tool will be an alternative to `osrm-contract`, using a different speedup algorithm.

Once the data is serialized to disk, `osrm-routed` and `libosrm` will need to be adapted to load this file and [use an alternative routing algorithm](https://github.com/Project-OSRM/osrm-backend/issues/3208) to route on the different data structure.

**Comment by james-gould** [comment]
Forgive me if this is the wrong place to ask but what source of traffic data will be used? Or isn't there a definite one yet?

**Comment by daniel-j-h** [comment]
There is no free and open traffic provider out there yet from what we can tell - and OSM does not contain real-time traffic data. This does not mean companies don't have such datasets available :)

**Comment by james-gould** [comment]
Waze might be something to look into, not sure of their position on open source contributions though :)

**Comment by TheMarex** [comment]
Output of customize should be a file `.overlay` that can be ingested by #3351 containing the weights and border vertices for the overlay graph.

**Comment by TheMarex** [comment]
This has landed with #3738.
